Sources/Kitura/RouterResponse.swift

```diff
@@ -520,22 +526,35 @@ extension RouterResponse {
             return self
         }
         do {
-            headers.setType("json")
+            headers["Content-Type"] = mediaType.description
             send(data: try encoder.encode(obj))
         } catch {
             Log.warning("Failed to encode Codable object for sending: \(error.localizedDescription)")
+            status(.internalServerError)
         }
 
         return self
     }
 
-    /// Send Encodable Object JSON Convienence Method
+    /// Send JSON Encodable Object
     ///
-    /// - Parameter json: the Encodable object to send.
+    /// - Parameter json: the JSON Encodable object to send.
     /// - Returns: this RouterResponse.
     @discardableResult
     public func send<T : Encodable>(json: T) -> RouterResponse {
-        return send(json)
+        guard !state.invokedEnd else {
+            Log.warning("RouterResponse send(_ obj:) invoked after end() for \(self.request.urlURL)")
+            return self
+        }
+        do {
+            headers.setType("json")
+            send(data: try JSONEncoder().encode(json))
```

Comment:
It would be confusing to the user if they customize their JSON encoder that we ignore it and use a default one here.